### PR TITLE
add note about file permissions

### PR DESCRIPTION
**Pull request by twpayne**
`ioutil.TempFile` on macOS creates files with mode `0600`, which is different to the default mode of `0777 &^ umask`. This PR adds a note to the documentation reminding the user to call `Chmod` if needed.

### CODE REVIEW

File: tempfile.go [hunk 1]
```diff
@@ -133,6 +133,10 @@ func (t *PendingFile) CloseAtomicallyReplace() error {
 // going to write a large number of files to the same file system, store the
 // result of TempDir(filepath.Base(path)) and pass it instead of the empty
 // string.
+//
+// The file's permissions may vary from system to system (e.g. 0600 on macOS).
```
Comment (stapelberg) [inline tempfile.go]: As per https://github.com/golang/go/blob/41e62b8c49d21659b48a95216e3062032285250f/src/io/ioutil/tempfile.go#L65, this is not actually system-dependent.

So, let’s rephrase this to state outright that the file will have 0600 unless Chmod is called please :)

Comment (twpayne) [reply tempfile.go]: Done.

File: tempfile.go [hunk 2]
```diff
@@ -133,6 +133,9 @@ func (t *PendingFile) CloseAtomicallyReplace() error {
 // going to write a large number of files to the same file system, store the
 // result of TempDir(filepath.Base(path)) and pass it instead of the empty
 // string.
+//
+// The file's permissions will be 0600 by default. You can change these by
+// explictly calling Chmod on the returned file.
```
Comment (stapelberg) [inline tempfile.go]: For clarity, can we replace “on the returned file” with “on the returned PendingFile” please?

Comment (twpayne) [reply tempfile.go]: Done.

**Comment by stapelberg** [comment]
Thanks!
